### PR TITLE
chore: fix template ids for project 

### PR DESCRIPTION
**Pull request by benPearce1**
This fixes the IDs on the templates attached to a project from getting new guid ID values everytime they are changed.

before: 
image

after:
image

### CODE REVIEW

File: octopusdeploy_framework/schemas/project.go [hunk 1]
```diff
@@ -129,7 +129,7 @@ func GetProjectResourceSchema() resourceSchema.Schema {
 			"template": resourceSchema.ListNestedBlock{
 				NestedObject: resourceSchema.NestedBlockObject{
 					Attributes: map[string]resourceSchema.Attribute{
-						"id":            util.ResourceString().Optional().Computed().Description("The ID of the template parameter.").Build(),
+						"id":            util.ResourceString().Optional().Computed().PlanModifiers(stringplanmodifier.UseStateForUnknown()).Description("The ID of the template parameter.").Build(),
```
Comment (benPearce1) [inline octopusdeploy_framework/schemas/project.go:132]: copy the value from state, if available

File: octopusdeploy_framework/resource_project_test.go [hunk 2]
```diff
@@ -114,17 +114,22 @@ func TestAccProjectWithUpdate(t *testing.T) {
...
 				),
-				Config: testAccProjectBasic(lifecycleLocalName, lifecycleName, projectGroupLocalName, projectGroupName, localName, name, description),
+				Config: testAccProjectBasic(lifecycleLocalName, lifecycleName, projectGroupLocalName, projectGroupName, localName, name, description, 3),
 			},
 		},
 	})
 }
 
-func testAccProjectBasic(lifecycleLocalName string, lifecycleName string, projectGroupLocalName string, projectGroupName string, localName string, name string, description string) string {
+func testAccProjectBasic(lifecycleLocalName string, lifecycleName string, projectGroupLocalName string, projectGroupName string, localName string, name string, description string, templateCount int) string {
```
Comment (benPearce1) [inline octopusdeploy_framework/resource_project_test.go:123]: this change allows the templates to be different between runs

File: octopusdeploy_framework/resource_project_expand.go [hunk 3]
```diff
@@ -138,7 +139,7 @@ func expandProject(ctx context.Context, model projectResourceModel) *projects.Pr
 }
 
 func expandGitLibraryPersistenceSettings(ctx context.Context, model gitLibraryPersistenceSettingsModel) projects.GitPersistenceSettings {
-	url, _ := url.Parse(model.URL.ValueString())
+	gitUrl, _ := url.Parse(model.URL.ValueString())
```
Comment (benPearce1) [inline octopusdeploy_framework/resource_project_expand.go:142]: code analysis was complaining about the name being the same as the imported package

File: octopusdeploy_framework/resource_project_expand.go [hunk 4]
```diff
@@ -74,9 +75,9 @@ func expandProject(ctx context.Context, model projectResourceModel) *projects.Pr
 		var gitAnonymousSettingsList []gitAnonymousPersistenceSettingsModel
 		diags := model.GitAnonymousPersistenceSettings.ElementsAs(ctx, &gitAnonymousSettingsList, false)
 		if diags.HasError() {
-			fmt.Printf("Error converting Git anonymous persistence settings: %v\n", diags)
+			tflog.Error(ctx, fmt.Sprintf("Error converting Git anonymous persistence settings: %v\n", diags))
```
Comment (benPearce1) [inline octopusdeploy_framework/resource_project_expand.go:78]: I modified these so they wrote out correctly to the terraform logs. In the tests these were writing to the console.

**Review by IsaacCalligeros95 — APPROVED** [review]
LGTM

**Pull request merged**